src/app.ts

```diff
@@ -172,8 +172,22 @@ export class App<State> {
     return this;
   }
 
-  use(middleware: MiddlewareFn<State>): this {
-    this.#router.addMiddleware(middleware);
+  use(
+    pathOrMiddleware: string | URLPattern | MiddlewareFn<State>,
+    middleware?: MiddlewareFn<State>,
+  ): this {
+    if (
+      typeof pathOrMiddleware === "string" ||
+      pathOrMiddleware instanceof URLPattern
+    ) {
+      // First argument is a path
+      if (!middleware) {
+        throw new Error("Middleware is required when path is provided");
+      }
+      this.#router.addMiddleware(pathOrMiddleware, middleware);
+    } else {
+      this.#router.addMiddleware("/*", pathOrMiddleware);
+    }
```

Comment:
```suggestion
    if (typeof pathOrMiddleware === "function") {
      this.#router.addMiddleware("/*", pathOrMiddleware);
      return this;
    }

    if (!middleware) {
      throw new TypeError("Middleware is required when path is provided");
    }
    this.#router.addMiddleware(pathOrMiddleware, middleware);
```
Maybe a little tidier?

Reply:
Thank you.
With the suggested approach, pathOrMiddleware can't be narrowed down to string | URLPattern, which causes a type error—so I decided to keep it as it is for now.